MultiplayerMod/Game/Mechanics/FilterEvents.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using HarmonyLib;
+using MultiplayerMod.Core.Patch;
+using MultiplayerMod.Multiplayer.Objects;
+
+namespace MultiplayerMod.Game.Mechanics;
+
+public static class FilterEvents {
+
+    public static event Action<MultiplayerReference, Tag> SetFilter;
+    public static event Action<MultiplayerReference, Tag> AddTagToFilter;
+    public static event Action<MultiplayerReference, Tag> RemoveTagFromFilter;
+
+    [HarmonyPatch(typeof(Filterable))]
+    private class FilterableEvents {
```

Comment:
You can use all these without subclasses, just `[HarmonyPatch(typeof(Filterable), nameof(Filterable.OnFilterChanged))]`

Reply:
Done